### ISSUE TITLE
Usernames with more than one word in them do not work

### ISSUE THREAD

**Issue by nthieberger** [nabu-catalog/nabu#392]
I tried using the name Anne Marie Hari, with Hari as the surname and the upload of a metadata sheet failed because it couldn't find the user. When I added a hyphen (Anne-Marie) it all worked ok. So, the current system doesn't allow >1 word in a name. Can/Should this be fixed or should we use workarounds like the hyphen?

**Comment by silviapfeiffer** [comment]
Probably should be fixed. Thanks for reporting the bug.

**Comment by JonoGillett** [comment]
So from a quick look at this it seems the issue isn't with 2-word names at all, rather that the importer doesn't currently allow contact-only collectors. It has an explicit check for "can I find an existing user with this name? no, then fail" whereas for other fields like item agents it creates a contact user if one can't be found.

What would you like the behaviour to be? I can make it behave the same and create a new contact-only user if it can't find a match.

**Comment by nthieberger** [comment]
I'll check on the issue you raise. But the problem of usernames with > 1
word is real and should be fixed so that 'Mary Anne' can be accepted as a
name for example.

**Comment by JonoGillett** [comment]
Could you give an example of where a 2-word name doesn't work? I've been using 'Anne Marie' in a few places for my testing and it seems to behave as expected.

**Comment by stasb** [comment]
I think I solved this issue. The collector name field in the XLS spreadsheet is parsed using spaces, so a name like 'Anne Marie Hari' would be parsed as first name 'Anne', and last name 'Marie Hari'.

To fix this I added a comma separator, so the cell must now be in the format 'Anne Marie, Hari', first name followed by last name. Any number of multiple names for first or last name can be used, as long as they are separated by a comma.

There is no inherent issue with saving multiple names in the database.
